packages/web-common/src/SessionStorageSessionManager.ts

```diff
@@ -0,0 +1,59 @@
+
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SessionManager } from "./types/SessionManager";
+
+const SESSION_STORAGE_KEY = 'opentelemetry-session-id';
+
+export class SessionStorageSessionManager implements SessionManager {
+  private _sessionId: string;
+
+  constructor() {
+    this._sessionId = this._getOrCreateSessionId();
+  }
+
+  getSessionId(): string {
+    return this._sessionId;
+  }
+
+  private _getOrCreateSessionId(): string {
+    let sessionId = sessionStorage.getItem(SESSION_STORAGE_KEY);
+    if (sessionId === null) {
+      const generateSessionId = getIdGenerator(16);
+      sessionId = generateSessionId();
+      sessionStorage.setItem(SESSION_STORAGE_KEY, sessionId);
```

Comment:
it may be worth also storing the created/last accessed time and allowing the manager optionally to be configured with how much time can pass before considering the old session to be expired, here's some of that being done on Android for reference: https://github.com/open-telemetry/opentelemetry-android/blob/main/android-agent/src/main/java/io/opentelemetry/android/SessionId.java